File: examples/rails/080_configuration/app/controllers/talker_controller.rb
class TalkerController < ActionController::API
  def say
    begin
      talker = Talker.find(0)
      render plain: talker.answer + ", " + talker.name + "!\n"
    rescue ActiveRecord::RecordNotFound => e
      render plain: "I have nothing to say.\n"
    end
  end

  def remember
    unless params.has_key?(:answer)
      render plain: "You forgot the answer :(\n" and return
    end
    unless params.has_key?(:name)
      render plain: "You forgot the name :(\n" and return
    end

    begin
      talker = Talker.find(0)
    rescue ActiveRecord::RecordNotFound => e
      talker = Talker.new
    end
    talker.update(id: 0, answer: params[:answer], name: params[:name])

    render plain: "Got it.\n"
  end
end

File: examples/laravel/050_s3/app/Http/Controllers/TalkerController.php
<?php

namespace App\Http\Controllers;

use App\Models\Talker;
use Illuminate\Http\Request;
use Illuminate\Routing\Controller;

class TalkerController extends Controller
{
    public function say(): string
    {
        /** @var Talker $talker */
        $talker = Talker::first();

        if (!$talker) {
            return "I have nothing to say";
        }

        return $talker->answer . ", " . $talker->name . "!";
    }

    public function remember(Request $request): string
    {
        if (!$request->has('answer')) {
            return "You forgot the answer :(";
        }

        if (!$request->has('name')) {
            return "You forgot the name :(";
        }

        /** @var Talker $talker */
        $talker = Talker::firstOrNew();
        $talker->answer = $request->input('answer');
        $talker->name = $request->input('name');
        $talker->save();

        return "Got it.";
    }
}
